pkg/namespace-labeler/labeler.go

```diff
@@ -29,25 +31,41 @@ import (
 type Labeler struct {
 	enforcedNsLabel       string
 	prometheusRuleLabeler bool
-	excludeList           map[string]map[string]struct{}
+	excludeList           map[string]map[string][]string
+	//deprecated - keep for backward compatibility
+	excludedRulesConfig map[string]map[string]struct{}
```

Comment:
It would be simpler if this package only expects `[]monitoringv1.ObjectReference` as input. Up to the caller (the prometheus or thanos controller) to convert from `[]monitoringv1.PrometheusRuleExcludeConfig` to `[]monitoringv1.ObjectReference`.